platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/internal/cc/impl/fingerprint/ConfigurationCacheFingerprintChecker.kt

```diff
@@ -235,18 +236,25 @@ class ConfigurationCacheFingerprintChecker(private val host: Host) {
                 when {
                     host.gradleUserHomeDir != gradleUserHomeDir -> text("Gradle user home directory has changed")
                     jvmFingerprint() != jvm -> text("JVM has changed")
-                    host.startParameterProperties != startParameterProperties -> text("the set of Gradle properties has changed")
-                    host.ignoreInputsInConfigurationCacheTaskGraphWriting != ignoreInputsInConfigurationCacheTaskGraphWriting -> text("the set of ignored configuration inputs has changed")
-                    host.instrumentationAgentUsed != instrumentationAgentUsed -> text("the instrumentation Java agent ${if (instrumentationAgentUsed) "is no longer available" else "is now applied"}")
-                    host.ignoredFileSystemCheckInputs != ignoredFileSystemCheckInputPaths -> text("the set of paths ignored in file-system-check input tracking has changed")
+                    host.startParameterProperties != startParameterProperties ->
+                        text("the set of Gradle properties has changed: ").text(detailedMessageForChanges(host.startParameterProperties, startParameterProperties))
+
+                    host.ignoreInputsInConfigurationCacheTaskGraphWriting != ignoreInputsInConfigurationCacheTaskGraphWriting ->
+                        text("the value of ignored configuration inputs flag (${StartParameterBuildOptions.ConfigurationCacheIgnoreInputsInTaskGraphSerialization.PROPERTY_NAME}) has changed")
+
+                    host.instrumentationAgentUsed != instrumentationAgentUsed ->
+                        text("the instrumentation Java agent ${if (instrumentationAgentUsed) "is no longer available" else "is now applied"}")
+
+                    host.ignoredFileSystemCheckInputs != ignoredFileSystemCheckInputPaths ->
+                        text("the set of paths ignored in file-system-check input tracking (${StartParameterBuildOptions.ConfigurationCacheIgnoredFileSystemCheckInputs.PROPERTY_NAME}) has changed")
                     else -> null
                 }
             }
 
             is ConfigurationCacheFingerprint.EnvironmentVariablesPrefixedBy -> input.run {
                 val current = System.getenv().filterKeysByPrefix(prefix)
                 ifOrNull(current != snapshot) {
-                    text("the set of environment variables prefixed by ").reference(prefix).text(" has changed")
+                    text("the set of environment variables prefixed by ").reference(prefix).text(" has changed: ").text(detailedMessageForChanges(snapshot, current))
```

Comment:
❓ Any chance this might end up exposing sensitive information in the console? 🤔

Reply:
It only prints keys, not values, so I doubt that